Add "How it works" H2s to 15 how-to guides

ptgott opened the pull request.

ptgott committed:
Add "How it works" H2s to 15 how-to guides

zmb3 commented on docs/pages/admin-guides/access-controls/idps/saml-grafana.mdx:
```diff
@@ -13,6 +13,15 @@ and Grafana to accept the identities it provides.
+Grafana enables users to authenticate using SAML. You can export SAML IdP
+metadata from Teleport, then provide it to your Grafana configuration file in
+order to instruct Grafana to trust the Teleport IdP's certificate authority.
+Your Teleport cluster then uses a SAML IdP service provider resource (a dynamic
```
Is this too low level? I don't know that readers are about dynamic resources and auth services backend.

I was going to suggest simplifying this section to just say something like "Grafana supports SAML IdPs, and Teleport can be a SAML IdP." Then I realized that's basically what we already said in the intro paragraph.

So the more I think about it, I'm feeling like our [rather arbitrary] requirement to put a _How it works_ section at the beginning of each guide is just causing us to write extra filler content that's either repetitive or too detailed for the intended audience.

ptgott replied:
For me, admins should be expected to be aware of Teleport dynamic resources and how to configure the behavior of a Teleport cluster by managing them. Since this is an admin guide, it makes sense to explain how this feature works in terms of managing dynamic resources. For a guide targeted to end users, like those in the "Connect your Client" section of the docs, it would make sense to remove all mention of dynamic resources.

Re: the "How it works" expectation in general, I think most of our how-to guides could use an architectural overview. Otherwise, I've found myself having to read through a whole guide to figure out what's going on, making the reading experience more frustrating. Before we started rolling out this section, how-to guides tended not to include any architectural overview at all.

zmb3 commented on docs/pages/admin-guides/access-controls/login-rules/guide.mdx:
```diff
@@ -6,8 +6,23 @@ labels:
-This guide will walk you through the process of writing, testing, and adding the
-first Login Rule to your Teleport cluster.
+Login Rules define logic that transforms the external traits of a user who signs
+into Teleport, providing consistent access control policies for users in your
```
A key use case for login rules is when you the team who manages Teleport is not the same as the team who manages the IdP. In this case, it's often easier to use login rules to make transformations to the traits prior to Teleport's SSO process than to request changes in the IdP configuration.

zmb3 commented on docs/pages/admin-guides/management/admin/daemon.mdx:
```diff
@@ -9,7 +9,15 @@ On Linux systems in non-containerized environments, we recommend running the `te
 configuration options that allow for better optimization, uptime, and availability.
 
-This guide will outline best practices for installing, configuring, and starting teleport as a daemon on a Linux host using systemd.
+This guide will outline best practices for installing, configuring, and starting `teleport` as a daemon on a Linux host using systemd.
```
Question - do we believe that this guide still covers the best practices for running Teleport on a Linux host?

I wonder if we think it's time to reflect that using `teleport-update` is considered best practice.

@hugoShaka knows best here.

hugoShaka replied:
Thank you @zmb3, yes `teleport-update` now takes care of installing and setting up the system service, this is how we prevent users from making mistakes. The recommended install should now be:
- run `curl https://teleport.example.com/scripts/install.sh | sudo bash`
- run `teleport configure`
- run `systemctl enable teleport`

zmb3 commented on docs/pages/zero-trust-access/infrastructure-as-code/teleport-operator/teleport-operator-standalone.mdx:
```diff
@@ -10,6 +10,22 @@ This guide explains how to run the Teleport Kubernetes Operator against any remo
+
+- A Teleport role
+- A join token
+- An Machine & Workload Identity Bot
```
```suggestion
- A Machine & Workload Identity Bot
```

ptgott committed:
Respond to zmb3 feedback

- Mention the Web UI and remove the attestation mention in the
  Passwordless "How it works" section.
- Remove some superfluous wording in the Grafana SAML guide "How it
  works" section.
- Mention a Login Rules use case.
